Frontend/app/problem/display/displayProblem.html

```diff
@@ -0,0 +1,74 @@
+<div class="row">
+  <div class="col-xs-12">
+    <h2>
+      Solving a Problem
+    </h2>
+    <p class="lead" *ngIf="problem">
+      Equation I'm trying to solve: <em>{{problem}}</em>
+    </p>
+  </div>
+</div>
+<!-- show if the problem has loaded -->
+<div *ngIf="!isLoading">
+  <div *ngFor="let step of problemTree; let goodStepIndex = index;">
+    <div class="row">
+      <div class="col-xs-12">
+        <div class="panel panel-default" *ngIf="goodStepIndex <= activeStepCount && !step.badStepShown">
```

Comment:
Because of the amount of logic in the HTML could comments please be added for each ngFor and ngIf?